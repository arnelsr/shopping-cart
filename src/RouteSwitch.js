import { BrowserRouter, Routes, Route, Link } from "react-router-dom";
import App from "./components/App";
import Shop from "./components/Shop";
import React, {useState} from 'react';

const RouteSwitch = () => {
    const[cart,setCart]=useState(0);
    const addCart=(e)=>{
        console.log('addcart');
        setCart(cart+1);
    }
  return (
    <BrowserRouter>
        <div className="App">
            <header className="App-header">
                <div className="App-header-1">
                    <Link to="/"><h1>Pet Shop</h1></Link>
                    <div className="App-header-right">
                        <Link to="/"><h2>Home</h2></Link>
                        <Link to="/shop"><h2>Shop</h2></Link>
                        <Link to="/"><h4 className="fa badge fa-lg" value={cart}>&#xf07a;</h4></Link>
                    </div>
                </div>
            </header>
            <Routes>
                <Route path="/" element={<App />} />
                <Route path="/shop" element={<Shop addCart={addCart} />} />
            </Routes>
            <footer className="App-footer">
                @arnelsr
            </footer>
        </div>
    </BrowserRouter>
  );
};

export default RouteSwitch;