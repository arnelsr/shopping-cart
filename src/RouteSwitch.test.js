import { render, screen } from '@testing-library/react';
import RouteSwitch from './RouteSwitch';

describe('Route Switch',()=>{
  it('renders snapshot',()=>{
    //container
    const {container}=render(<RouteSwitch/>);
    expect(container).toMatchSnapshot();
  });
});


