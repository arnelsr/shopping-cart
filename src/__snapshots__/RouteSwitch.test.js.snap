// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Switch renders snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="App-header-1"
      >
        <a
          href="/"
        >
          <h1>
            Pet Shop
          </h1>
        </a>
        <div
          class="App-header-right"
        >
          <a
            href="/"
          >
            <h2>
              Home
            </h2>
          </a>
          <a
            href="/shop"
          >
            <h2>
              Shop
            </h2>
          </a>
          <h2>
            Cart:
          </h2>
          <h2>
            0
          </h2>
        </div>
      </div>
    </header>
    <div
      class="App-content"
    >
      <div>
        <h1>
          Get Your New
        </h1>
        <h1>
          Pet Here!
        </h1>
      </div>
      <img
        alt="Dog"
        height="600"
        src="dog.png"
        width="600"
      />
    </div>
    <footer
      class="App-footer"
    >
      @arnelsr
    </footer>
  </div>
</div>
`;
